components/apps.js

```diff
@@ -649,33 +647,147 @@ SteamUser.prototype.getOwnedDepots = function(excludeSharedLicenses) {
+ * @param {object|function} filter - Options for what counts for ownership, or a custom filter function
  * @returns {boolean}
  */
-SteamUser.prototype.ownsDepot = function(depotid, excludeSharedLicenses) {
-	return this.getOwnedDepots(excludeSharedLicenses).indexOf(parseInt(depotid, 10)) != -1;
+SteamUser.prototype.ownsDepot = function(depotid, filter) {
+	return this.getOwnedDepots(filter).indexOf(parseInt(depotid, 10)) != -1;
 };
 
 /**
- * Returns an array of package IDs this account owns. Only works if enablePicsCache option is enabled and appOwnershipCached event
- * has been emitted.
- * @param {boolean} [excludeSharedLicenses=false] - Pass true to exclude licenses that we have through family sharing
+ * Returns an array of licenses this account owns.
  * @returns {int[]}
  */
-SteamUser.prototype.getOwnedPackages = function(excludeSharedLicenses) {
-	if (this.steamID.type != SteamID.Type.ANON_USER && !this.licenses) {
+SteamUser.prototype.getOwnedLicenses = function() {
+	if (!this.licenses) {
 		throw new Error("We don't have our license list yet.");
 	}
 
+	return this.licenses;
+};
+
+/**
+ * Returns an array of package IDs this account owns (different from owned licenses). The filter only
+ * works, if enablePicsCache option is enabled and appOwnershipCached event has been emitted.
+ * @param {object|function} filter - Options for what counts for ownership, or a custom filter function
+ * @returns {int[]}
+ */
+SteamUser.prototype.getOwnedPackages = function(filter) {
+	// We're anonymous
 	if (this.steamID.type == SteamID.Type.ANON_USER) {
 		return [17906];
 	}
 
 	// We're an individual user
-	let packages = this.licenses;
-	if (excludeSharedLicenses) {
-		packages = packages.filter(license => license.owner_id == this.steamID.accountid);
+	let packages = this.getOwnedLicenses();
+
+	// If passed filter argument is empty, and global ownership filter is set, use it
+	if (!filter && this.options.ownershipFilter) {
+		filter = this.options.ownershipFilter;
 	}
+
+	// Support for the old syntax and behavior (<= v4.19.12)
+	if (!filter) {
+		// Slight deviation of old behavior: don't return expired licenses
+		return this._returnPackages(packages);
+	}
+	// If filter options is a boolean, we asssume it's excludeSharedLicenses
+	if (typeof filter === 'boolean') {
+		let excludeSharedLicenses = filter;
+		if (excludeSharedLicenses) {
+			packages = packages.filter(license => license.owner_id == this.steamID.accountid);
+		}
+		// Slight deviation of old behavior: don't return expired licenses
+		return this._returnPackages(packages);
+	}
+
+	// New behavior from this point on
+	// Handle filter options
+	if (typeof filter === 'object') {
+		const defaults = {
+			excludeFree: false, // By default, include free licenses (FreeOnDemand, NoCost, etc.)
+			excludeShared: true, // By default, exclude shared licenses
+			excludeExpiring: true // By default, exclude licenses that are going to expire in the future (free weekends)
```

Comment:
I would prefer to have all exclusions be false by default. I understand that this isn't a backwards-compatibility issue since this logic only kicks in if you pass a filter options object, but it makes more intuitive sense to me that you'd need to manually specify which classes of license you want to exclude without anything defaulting to excluded.